frontend/packages/console-dynamic-plugin-sdk/src/app/core/reducers/coreSelector.ts

```diff
@@ -0,0 +1,22 @@
+import { ImpersonateKind, RootState, UserKind } from '../../redux-types';
+
+type GetImpersonate = (state: RootState) => { impersonate: ImpersonateKind };
+type GetUser = (state: RootState) => { user: UserKind };
+
+/**
+ * It provides impersonation details from the redux store.
+ * @param state the root state
+ * @return The the impersonate state.
+ * * */
+export const getImpersonate: GetImpersonate = (state) => {
+  return { impersonate: state.core.impersonate };
```

Comment:
Oh hey, you have selectors! -- why are you not using them & why are they object responses?

Reply:
yeah can use them directly, make sense. Thanks :)